### PR TITLE
[Fixes #852] XmlDictionaryReader.CreateTextReader overload which support...

### LINKED ISSUE #852: XmlDictionaryReader.CreateTextReader overload which supports Encoding is missing
This overload is missing in ProjectK:
```c#
public static XmlDictionaryReader CreateTextReader(
	Stream stream,
	Encoding encoding,
	XmlDictionaryReaderQuotas quotas,
	OnXmlDictionaryReaderClose onClose
)
```
Link: http://msdn.microsoft.com/en-us/library/aa346768(v=vs.110).aspx

This is used by ASP .NET vNext during Deserialization of Xml input.

### PR DESCRIPTION
**Pull request by kichalla**
...s Encoding is missing

@sornaks @rynowak

### CODE REVIEW

**Comment by kichalla** [comment]
ping

File: test/Microsoft.AspNet.Mvc.Xml.Test/XmlDataContractSerializerInputFormatterTest.cs [hunk 1]
```diff
@@ -296,7 +296,7 @@ public void SetMaxDepth_ThrowsWhenMaxDepthIsBelowOne()
                                                                 typeof(XmlException);
             var expectedMessage = TestPlatformHelper.IsMono ?
                 "Expected element 'TestLevelTwo' in namespace '', but found Element node 'DummyClass' in namespace ''" :
-                "The encoding in the declaration 'UTF-8' does not match the encoding of the document 'utf-16LE'.";
+                "The expected encoding 'utf-8' does not match the actual encoding 'utf-16LE'.";
```
Comment (sornaks) [inline test/Microsoft.AspNet.Mvc.Xml.Test/XmlDataContractSerializerInputFormatterTest.cs]: Can you add a test case showing that the encoding in `charset` overrides the one in `<?xml version=\"1.0\" encoding=\"UTF-8\"?>`?

Comment (kichalla) [inline test/Microsoft.AspNet.Mvc.Xml.Test/XmlDataContractSerializerInputFormatterTest.cs]: Sure :+1:

**Comment by sornaks** [comment]
:watch:

**Comment by kichalla** [comment]
Updated. /cc: @sornaks

**Comment by kichalla** [comment]
Update. /cc: @sornaks

**Comment by sornaks** [comment]
:shipit:

**Pull request merged**